src/sources/opentelemetry/mod.rs

```diff
@@ -37,17 +60,34 @@ pub struct OpentelemetryConfig {
     #[configurable(derived)]
     #[serde(default)]
     tls: Option<TlsEnableableConfig>,
+}
+
+/// Configuration for the `opentelemetry` HTTP server.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+struct HttpConfig {
+    /// The address to listen for connections on.
+    ///
+    /// It _must_ include a port.
```

Comment:
What happens if it doesn't ? Is that something that should be explicitly checked for ?

Reply:
```text
2022-08-02T20:26:24.412426Z ERROR vector::cli: Configuration error. error=invalid socket address syntax for key `sources.otel`
```

We don't check this in either `datadog_agent` or the `util::http::prelude` servers. I'd rather include that in an improved shared server we can reuse more easily.